src/wallets/browser/NearWallet.ts

```diff
@@ -20,14 +20,21 @@ export default class NearWallet extends BrowserWallet implements INearWallet {
   }
 
   async walletSelected() {
-    this.signIn();
+    await this.signIn();
   }
 
   async init() {
     const state = getState();
     if (!state.nearConnection) return;
     this.wallet = new WalletConnection(state.nearConnection, "near_app");
     EventHandler.callEventHandler("init");
+    if (!this.wallet.isSignedIn()) {
+      updateState((prevState) => ({
+        ...prevState,
+        signedInWalletId: null,
+        isSignedIn: false,
+      }));
+    }
```

Comment:
I'm concerned we're trying to treat the symptoms here instead of fixing it at the source. Had a quick look over the documentation, but it doesn't really give us much on the nuances to `wallet.requestSignIn`. I'm wondering if we should modify the `signIn` method (which actually isn't really async as the Promise isn't returned) to look something like:

```ts
const state = getState();
this.wallet.requestSignIn(state.options.contract.address).then(() => {
  // My thinking is this block should trigger when the user
  // cancels since they didn't actually sign in.
  if (!this.wallet.isSignedIn()) { 
    return;
  }

  this.setWalletAsSignedIn();
  EventHandler.callEventHandler("signIn");
});
```

Reply:
I have followed the suggestion to add the check inside the `then` block of the `requestSignIn` method, then there was another issue with updating the state which we resolved together in a call.
I will push the changes in a bit, Thanks for the help earlier.